src/test/java/io/jenkins/plugins/coverage/QualityGateEvaluatorTest.java

```diff
@@ -0,0 +1,233 @@
+        when(coverageNode.getMetricPercentages()).thenReturn((SortedMap<CoverageMetric, Double>) coverage);
+
+
+        QualityGateStatus qualityGateStatus = qualityGateEvaluator.evaluate(coverageNode, logger);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.WARNING);
+    }
+
+    @Test
+    void testMultipleQualityGatesShouldFail(){
+        Logger logger = new Logger();
+        QualityGateEvaluator qualityGateEvaluator = new QualityGateEvaluator();
+
+        QualityGate qualityGate1 = new QualityGate(CoverageMetric.LINE, 50.0, 5.0);
+        QualityGate qualityGate2 = new QualityGate(CoverageMetric.BRANCH, 50.0, 5.0);
+        qualityGateEvaluator.add(qualityGate1);
+        qualityGateEvaluator.add(qualityGate2);
+
+
+        CoverageNode coverageNode = mock(CoverageNode.class);
+        Map<CoverageMetric, Double> coverage = new TreeMap<CoverageMetric, Double>();
+        coverage.put(CoverageMetric.LINE, 51.0);
+        coverage.put(CoverageMetric.BRANCH, 5.0); // only Branch Coverage fails, still results in failing
+        when(coverageNode.getMetricPercentages()).thenReturn((SortedMap<CoverageMetric, Double>) coverage);
+
+
+        QualityGateStatus qualityGateStatus = qualityGateEvaluator.evaluate(coverageNode, logger);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.FAILED);
+    }
+
+
+    @Test
+    void testMultipleQualityGatesShouldPass(){
+        Logger logger = new Logger();
+        QualityGateEvaluator qualityGateEvaluator = new QualityGateEvaluator();
+
+        QualityGate qualityGate1 = new QualityGate(CoverageMetric.LINE, 50.0, 5.0);
+        QualityGate qualityGate2 = new QualityGate(CoverageMetric.BRANCH, 50.0, 5.0);
+        qualityGateEvaluator.add(qualityGate1);
+        qualityGateEvaluator.add(qualityGate2);
+
+
+        CoverageNode coverageNode = mock(CoverageNode.class);
+        Map<CoverageMetric, Double> coverage = new TreeMap<CoverageMetric, Double>();
+        coverage.put(CoverageMetric.LINE, 51.0);
+        coverage.put(CoverageMetric.BRANCH, 51.0);
+        when(coverageNode.getMetricPercentages()).thenReturn((SortedMap<CoverageMetric, Double>) coverage);
+
+
+        QualityGateStatus qualityGateStatus = qualityGateEvaluator.evaluate(coverageNode, logger);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.PASSED);
+    }
+
+
+
+
+    /**
+     * Logger for the tests that provides a way verify and clear the messages.
+     */
+    private static class Logger implements FormattedLogger {
+        private final List<String> messages = new ArrayList<>();
+
+        @Override
+        public void print(final String format, final Object... args) {
+            messages.add(String.format(format, args));
+        }
+
+        List<String> getMessages() {
+            return messages;
+        }
+
+        void clear() {
+            messages.clear();
+        }
+    }
+
+}
```

Comment:
In total, the tests are well written and cover most important paths. (One is missing: a quality gate that has no match in the node). Some tests can be combined and assertions are missing for the logger.